adr/ADR-00.md

```diff
@@ -0,0 +1,42 @@
+# Metadata for Stream and Consumer
+
+|Metadata|Value|
+|--------|-----|
+|Date    |2023-01-23|
+|Author  |@Jarema|
+|Status  |Proposed|
+|Tags    |jetstream, client, server|
+
+## Context and Problem Statement
+
+Until now, there was no way to easily add additional information about Stream or Consumer.
+The only solution was using `Description` field, which is a not ergonomic workaround.
+
+## Server PR
+https://github.com/nats-io/nats-server/pull/3797
+
+## Design
+
+The solution is to add new `metadata` field to both `Consumer` and `Stream` config.
+The `metadata` field would by a map of `string` keys and `string` values.
+
+The map would be represented in json as object with nested key/value pairs, which is a default
+way to marshal maps/hashmaps in most languages.
+
+To avoid abuse of the metadata, the size of it will be limited to 1MB.
```

Comment:
1MB is very big - given nats messages are capped at 1MB, I'd aim for way lower.  1MB string:string map is huge

Reply:
I'm fine with clamping to 512kB or even 128kB.

@aricart any thoughts here?

Reply:
@wallyqs I guess even 128kB would be more than enough for anything orchestration related, like k8s.

Reply:
Afaik these are k8s label limits for comparison ```Each resource can have multiple cluster labels, up to a maximum of 64. Each cluster label must be a key-value pair. Keys have a minimum length of 1 character and a maximum length of 63 characters, and cannot be empty. Values can be empty, and have a maximum length of 63 characters.```

Reply:
128kb sounds more than enough, similar limits as k8s works too.

Reply:
Ok, 128k it is.

Reply:
128K sounds good to me as well